cmd/operator-sdk/scorecard/cmd.go

```diff
@@ -65,5 +78,107 @@ func NewCmd() *cobra.Command {
 	if err := viper.BindPFlag("scorecard."+scorecard.BundleOpt, scorecardCmd.Flags().Lookup(scorecard.BundleOpt)); err != nil {
 		log.Fatalf("Unable to add bundle :%v", err)
 	}
+
 	return scorecardCmd
 }
+
+func initConfig() (*viper.Viper, error) {
+	// viper/cobra already has flags parsed at this point; we can check if a config file flag is set
+	if viper.GetString(scorecard.ConfigOpt) != "" {
+		// Use config file from the flag.
+		viper.SetConfigFile(viper.GetString(scorecard.ConfigOpt))
+	} else {
+		viper.AddConfigPath(projutil.MustGetwd())
+		// using SetConfigName allows users to use a .yaml, .json, or .toml file
+		viper.SetConfigName(scorecard.DefaultConfigFile)
+	}
+
+	var scViper *viper.Viper
+	if err := viper.ReadInConfig(); err == nil {
+		scViper = viper.Sub("scorecard")
+		// this is a workaround for the fact that nested flags don't persist on viper.Sub
+		scViper.Set(scorecard.OutputFormatOpt, viper.GetString("scorecard."+scorecard.OutputFormatOpt))
+		scViper.Set(scplugins.KubeconfigOpt, viper.GetString("scorecard."+scplugins.KubeconfigOpt))
+		scViper.Set(schelpers.VersionOpt, viper.GetString("scorecard."+schelpers.VersionOpt))
+		scViper.Set(scorecard.SelectorOpt, viper.GetString("scorecard."+scorecard.SelectorOpt))
+		scViper.Set(scorecard.BundleOpt, viper.GetString("scorecard."+scorecard.BundleOpt))
+		scViper.Set(scorecard.ListOpt, viper.GetString("scorecard."+scorecard.ListOpt))
+		// configure logger output before logging anything
+		if !scViper.IsSet(scorecard.OutputFormatOpt) {
+			scViper.Set(scorecard.OutputFormatOpt, scorecard.TextOutputFormat)
+		}
+		format := scViper.GetString(scorecard.OutputFormatOpt)
+		if format == scorecard.TextOutputFormat {
+			scorecard.LogReadWriter = os.Stdout
+		} else if format == scorecard.JSONOutputFormat {
+			scorecard.LogReadWriter = &bytes.Buffer{}
+		} else {
+			return nil, fmt.Errorf("invalid output format: %s", format)
+		}
+		scorecard.Log.SetOutput(scorecard.LogReadWriter)
+		if err != nil {
```

Comment:
What error is being checked here?

Reply:
nice catch...must have been a bad copy/paste...will remove.